code/modules/atmospherics/machinery/components/unary_devices/passive_vent.dm

```diff
@@ -10,18 +10,28 @@
 	can_unwrench = TRUE
 
 	var/volume = 250
+	on = TRUE
 
 /obj/machinery/atmospherics/unary/passive_vent/high_volume
 	name = "large passive vent"
 	volume = 1000
 
+/obj/machinery/atmospherics/unary/passive_vent/high_volume/shadow
+	on = FALSE
+
+/obj/machinery/atmospherics/unary/passive_vent/high_volume/shadow/onTransitZ(old_z, new_z)
+	. = ..()
+	if(is_station_level(new_z))
+		on = TRUE
+
 /obj/machinery/atmospherics/unary/passive_vent/Initialize(mapload)
 	. = ..()
 	air_contents.volume = volume
 
 /obj/machinery/atmospherics/unary/passive_vent/process_atmos()
 	..()
-
+	if(!on)
+		return
```

Comment:
I dont think this should be considered for all vents, it could lead to some weird cases, can you put it on `/obj/machinery/atmospherics/unary/passive_vent/high_volume/shadow/process_atmos()`?

Reply:
Can look into that!

Reply:
done